hadoop-hdds/framework/src/main/java/org/apache/hadoop/hdds/security/token/BlockTokenVerifier.java

```diff
@@ -77,21 +75,20 @@ protected Object getService(ContainerCommandRequestProto cmd) {
 
   @Override
   protected void verify(OzoneBlockTokenIdentifier tokenId,
-      ContainerCommandRequestProto cmd) throws SCMSecurityException {
+      ContainerCommandRequestProtoOrBuilder cmd) throws SCMSecurityException {
 
-    ContainerProtos.Type type = cmd.getCmdType();
-    if (type == ReadChunk || type == GetBlock || type == GetSmallFile) {
-      if (!tokenId.getAccessModes().contains(READ)) {
-        throw new BlockTokenException("Block token with " + tokenId.getService()
-            + " doesn't have READ permission");
-      }
-    } else if (type == WriteChunk || type == PutBlock || type == PutSmallFile) {
-      if (!tokenId.getAccessModes().contains(WRITE)) {
-        throw new BlockTokenException("Block token with " + tokenId.getService()
-            + " doesn't have WRITE permission");
-      }
+    HddsProtos.BlockTokenSecretProto.AccessModeProto accessMode;
+    if (HddsUtils.isReadOnly(cmd)) {
+      accessMode = READ;
+    } else if (cmd.getCmdType() == DeleteBlock ||
+        cmd.getCmdType() == DeleteChunk) {
+      accessMode = DELETE;
```

Comment:
I did not find any place where we add block tokens with DELETE mode? Do you plan to add those in follow up JIRAs? I assume those will be used for some debug CLI as the deletion done async via Hadoop RPC.

Reply:
Thanks @xiaoyuyao for flagging this.  Indeed, further change is needed in SCM.

Reply:
I do not _plan_ to work on debug CLI.  Maybe someone will find that useful to take initiative.  But checking for the token still prevents user from sending such command without auth.

Regarding the further change I mentioned, I opened [HDDS-5264](https://issues.apache.org/jira/browse/HDDS-5264) and submitted another PR.  It turned out to be quite separate from this one, so on second thought I wanted to avoid mixing the two.

Reply:
bq. Regarding the further change I mentioned, I opened HDDS-5264 and submitted another PR. It turned out to be quite separate from this one, so on second thought I wanted to avoid mixing the two.

Sounds good to me. +1.